### PR TITLE
Add an environment variable which can skip compiling the ANA extension

### PR DESCRIPTION
**Pull request by Cadair**
This is to help out in https://github.com/holzschu/Carnets/issues/226

### CODE REVIEW

File: sunpy/io/setup_package.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 def get_extensions():
 
-    if get_compiler() == 'msvc':
+    if get_compiler() == 'msvc' or os.environ.get("SUNPY_NO_BUILD_ANA_EXTENSION"], None):
```
Comment (holzschu) [inline sunpy/io/setup_package.py:12]: I think you have an extra "]" here.

**Review by ConorMacBride — APPROVED** [review]
I ran `python -m build --wheel .` on this PR and it built `sunpy-4.1.dev95+g7d4ebe66c-cp39-cp39-macosx_10_9_x86_64.whl`, while running `SUNPY_NO_BUILD_ANA_EXTENSION=1 python -m build --wheel .` built `sunpy-4.1.dev95+g7d4ebe66c-py3-none-any.whl` so this seems to work.

I guess sunpy isn't going to add more extensions anytime soon, but would it be better to use `SUNPY_NO_BUILD_EXTENSIONS` and check in `setup.py` instead?

**Comment by nabobalis** [comment]
> I ran `python -m build --wheel .` on this PR and it built `sunpy-4.1.dev95+g7d4ebe66c-cp39-cp39-macosx_10_9_x86_64.whl`, while running `SUNPY_NO_BUILD_ANA_EXTENSION=1 python -m build --wheel .` built `sunpy-4.1.dev95+g7d4ebe66c-py3-none-any.whl` so this seems to work.
> 
> I guess sunpy isn't going to add more extensions anytime soon, but would it be better to use `SUNPY_NO_BUILD_EXTENSIONS` and check in `setup.py` instead?

I feel like we would want:
1. Per compiled flag, 
2. Overall complied flag

Doesn't have to be this PR.

**Review by nabobalis — APPROVED** [review]

**Comment by Cadair** [comment]
I would rather not promise that any future C extensions will be optional.

**Comment by nabobalis** [comment]
Cherry-picked to the 4.0 branch

**Pull request merged**